tests/test_poly.py

```diff
@@ -0,0 +1,287 @@
+import numpy
+import piff
+PolynomialsTypes = piff.POLYNOMIAL_TYPES.keys()
+
+
+
+def test_poly_indexing():
+    #Some indexing tests for a polynomial up to order 3
+    N = 3
+    interp = piff.Polynomial([N])
+
+    #We expect there to be these coefficients:
+    # x^0 y^0   1
+
+    # x^0 y^1   2
+    # x^1 y^0   3
+
+    # x^0 y^2   4
+    # x^1 y^1   5 
+    # x^2 y^0   6
+
+    # x^0 y^3   7
+    # x^1 y^2   8
+    # x^2 y^1   9
+    # x^3 y^0   10 
+
+    #Check that we have the indices we expect
+    assert interp.indices[0] == [
+        (0,0),
+        (0,1), (1,0),
+        (0,2), (1,1), (2,0),
+        (0,3), (1,2), (2,1), (3,0)
+    ]
+    assert interp.nvariables[0]==10
+
+    #check the packing then unpacking a 
+    packed = numpy.random.uniform(size=interp.nvariables[0])
+    unpacked = interp.unpack_coefficients(0,packed)
+    packed_test = interp.pack_coefficients(0,unpacked)
+
+    #Check that the shape is 4*4 in the unpacked (because we
+    #want space for all the terms), and that we can unpack and
+    #repack successfully.
+    assert (packed==packed_test).all()
+    assert unpacked.shape == (N+1,N+1)
+
+    unpacked_test = numpy.zeros_like(unpacked)
+
+    #check that we have zeros for the terms that should be zero in the matrix.
+    #We don't want any terms with total exponent > N.
+    for i in xrange(N+1):
+        for j in xrange(N+1):
+            if i+j>3:
+                assert unpacked[i,j]==0.0
+                unpacked_test[i,j]=0.0
+
+    #Now do the test the other way around, checking that 
+    #we can pack and then unpack
+    packed_test_2 = interp.pack_coefficients(0,unpacked_test)
+    unpacked_test_2 = interp.unpack_coefficients(0,packed_test_2)
+    assert (unpacked_test_2 == unpacked_test).all()
```

Comment:
This is testing that the polynomial coefficients with total order i+j greater than the maximum total order 3 are zero. It's testing unpacked[i,j]==0, which is not identically true since it's unpacked_test that was created just above with zeros_like, rather than unpacked; unpacked was created on line 38 by unpacked a random vector.